File: src/main/kotlin/ru/proshik/english/quizlet/telegramBot/repository/model/Account.java
package ru.proshik.english.quizlet.telegramBot.repository.model;

import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Parameter;

import javax.persistence.*;
import java.time.ZonedDateTime;

import static javax.persistence.GenerationType.SEQUENCE;
import static org.hibernate.id.enhanced.SequenceStyleGenerator.SEQUENCE_PARAM;

@Entity
@Table(name = "account")
public class Account {

    @Id
    @GeneratedValue(strategy = SEQUENCE, generator = "account_id_seq")
    @GenericGenerator(name = "account_id_seq",
            strategy = "enhanced-sequence",
            parameters = @Parameter(name = SEQUENCE_PARAM, value = "account_id_seq"))
    private Long id;

    private ZonedDateTime createdDate;

    private String login;

    private String accessToken;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    public Account() {
    }

    public Account(ZonedDateTime createdDate, String login, String accessToken) {
        this.createdDate = createdDate;
        this.login = login;
        this.accessToken = accessToken;
    }

    public Long getId() {
        return id;
    }

    public ZonedDateTime getCreatedDate() {
        return createdDate;
    }

    public String getLogin() {
        return login;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: src/main/kotlin/ru/proshik/english/quizlet/telegramBot/repository/model/Account.kt
package ru.proshik.english.quizlet.telegramBot.repository.model

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Parameter
import org.hibernate.id.enhanced.SequenceStyleGenerator
import java.time.ZonedDateTime
import javax.persistence.*


@Entity
@Table(name = "account")
data class Account(
        @Id
        @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "users_id_seq")
        @GenericGenerator(name = "users_id_seq",
                strategy = "enhanced-sequence",
                parameters = [Parameter(name = SequenceStyleGenerator.SEQUENCE_PARAM, value = "users_id_seq")])
        val id: Long? = null,

        val createdDate: ZonedDateTime,

        val login: String,

        @Column(name = "access_token")
        val accessToken: String,

        @OneToOne @MapsId
        var user: User?
)
